src/main/java/org/opensearch/neuralsearch/search/collector/HybridTopFieldDocSortCollector.java

```diff
@@ -0,0 +1,411 @@
+            int slot = hitsCollected - 1;
+            // Copy hit into queue
+            comparators[subQueryNumber].copy(slot, doc);
+            add(slot, doc, compoundScores[subQueryNumber], subQueryNumber, score);
+            if (queueFull[subQueryNumber]) {
+                comparators[subQueryNumber].setBottom(bottom.slot);
+            }
+        }
+
+        /*
+        // This hit is competitive - replace bottom element in queue & adjustTop
+         */
+        void collectCompetitiveHit(int doc, int subQueryNumber) throws IOException {
+            comparators[subQueryNumber].copy(bottom.slot, doc);
+            updateBottom(doc, compoundScores[subQueryNumber]);
+            comparators[subQueryNumber].setBottom(bottom.slot);
+        }
+
+        boolean thresholdCheck(int doc, int subQueryNumber) throws IOException {
+            if (collectedAllCompetitiveHits || reverseMul * comparators[subQueryNumber].compareBottom(doc) <= 0) {
+                // since docs are visited in doc Id order, if compare is 0, it means
+                // this document is larger than anything else in the queue, and
+                // therefore not competitive.
+                if (searchSortPartOfIndexSort) {
+                    if (hitsThresholdChecker.isThresholdReached()) {
+                        setTotalHitsRelation(TotalHits.Relation.GREATER_THAN_OR_EQUAL_TO);
+                        throw new CollectionTerminatedException();
+                    } else {
+                        collectedAllCompetitiveHits = true;
+                    }
+                }
+                return true;
+            }
+            return false;
+        }
+
+    }
+
+    @Override
+    public ScoreMode scoreMode() {
+        return hitsThresholdChecker.scoreMode();
+    }
+
+    /*
+      TopFieldDocs per subquery
+     */
+    protected TopFieldDocs topDocsPerQuery(
+        int start,
+        int howMany,
+        PriorityQueue<FieldValueHitQueue.Entry> pq,
+        int totalHits,
+        SortField[] sortFields
+    ) {
+        if (howMany < 0) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ROOT, "Number of hits requested must be greater than 0 but value was %d", howMany)
+            );
+        }
+
+        if (start < 0) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ROOT, "Expected value of starting position is between 0 and %d, got %d", howMany, start)
+            );
+        }
+
+        if (start >= howMany || howMany == 0) {
+            return EMPTY_TOPDOCS;
+        }
+
+        int size = howMany - start;
+        ScoreDoc[] results = new ScoreDoc[size];
+        // pq's pop() returns the 'least' element in the queue, therefore need
+        // to discard the first ones, until we reach the requested range.
+        for (int i = pq.size() - start - size; i > 0; i--) {
+            pq.pop();
+        }
```

Comment:
help me understand what you are trying to do here:
You are trying to reduce the size of the PQ by discarding the elements but one thing I am not getting is why we are doing that?

Reply:
Removed this as we do not need it.